### PR TITLE
weather service refactor precision Fixes #1312

### LINKED ISSUE #1312: weather service doesn't have enough precision
the weather  temp received for a location can look like this 
> 70 70 70 70 70 68 68 68 70 70

a more accurate source might be needed formonitoring the effects of upcoming eclipse in August

openweathermap.com/api

### PR DESCRIPTION
**Pull request by hamidzr**

### CODE REVIEW

File: src/server/rpc/procedures/weather/weather.js [hunk 1]
```diff
@@ -23,189 +21,113 @@ var isWithinMaxDistance = function(result, lat, lng) {
...
+        .then(body => {
             var temp = 'unknown';
             if (body.main && isWithinMaxDistance(body, latitude, longitude)) {
                 temp = body.main.temp;
                 trace('Kelvin temp is '+temp+' fahrenheit is '+tuc.k2f(temp));
-                temp = Math.round(tuc.k2f(temp));
+                temp = tuc.k2f(temp).toFixed(1);
             }
-            return response.json(temp);
+            this.response.send(temp);
```
Comment (brollb) [inline src/server/rpc/procedures/weather/weather.js]: why did you change this from `json`? This was using `json` before to make sure express didn't mistake it for a status code... I am guessing it will be ok when the number is not an integer but what if it rounds to `101.0` or something?

I think it would be safest not to change this...

File: src/server/rpc/procedures/weather/weather.js [hunk 2]
```diff
@@ -23,189 +21,113 @@ var isWithinMaxDistance = function(result, lat, lng) {
...
-            body = JSON.parse(body);
+weather.humidity = function(latitude, longitude){
+    return this._requestData({queryString: '&lat=' + latitude + '&lon=' + longitude})
+        .then(body => {
             var humidity = 'unknown';
             if (isWithinMaxDistance(body, latitude, longitude)) {
                 humidity = body.main.humidity;
             }
-            return response.json(humidity);
+            this.response.send(humidity);
```
Comment (brollb) [inline src/server/rpc/procedures/weather/weather.js]: same as above

File: src/server/rpc/procedures/weather/weather.js [hunk 3]
```diff
@@ -23,189 +21,113 @@ var isWithinMaxDistance = function(result, lat, lng) {
...
-            body = JSON.parse(body);
+weather.description = function(latitude, longitude){
+    return this._requestData({queryString: '&lat=' + latitude + '&lon=' + longitude})
+        .then(body => {
             var description = 'unknown';
             if (isWithinMaxDistance(body, latitude, longitude)) {
                 description = body.weather[0].description;
             }
-            return response.json(description);
+            this.response.send(description);
```
Comment (brollb) [inline src/server/rpc/procedures/weather/weather.js]: same as above

File: src/server/rpc/procedures/weather/weather.js [hunk 4]
```diff
@@ -23,189 +21,113 @@ var isWithinMaxDistance = function(result, lat, lng) {
...
-        var url = baseUrl + '&lat=' + latitude + '&lon=' + longitude,
-            response = this.response;
+weather.windSpeed = function(latitude, longitude){
+    return this._requestData({queryString: '&lat=' + latitude + '&lon=' + longitude})
+        .then(body => {
+            var speed = 'unknown';
+            if (isWithinMaxDistance(body, latitude, longitude)) {
+                speed = body.wind.speed || 'unknown';
+            }
+            this.response.send(speed);
```
Comment (brollb) [inline src/server/rpc/procedures/weather/weather.js]: same as above

File: src/server/rpc/procedures/weather/weather.js [hunk 5]
```diff
@@ -23,189 +21,113 @@ var isWithinMaxDistance = function(result, lat, lng) {
...
-                return response.status(500).send('ERROR: '+(err || body));
+weather.windAngle = function(latitude, longitude){
+    return this._requestData({queryString: '&lat=' + latitude + '&lon=' + longitude})
+        .then(body => {
+            var deg = 'unknown';
+            if (isWithinMaxDistance(body, latitude, longitude)) {
+                deg = body.wind.deg || 'unknown';
             }
-            body = JSON.parse(body);
+            this.response.send(deg);
```
Comment (brollb) [inline src/server/rpc/procedures/weather/weather.js]: same as above

File: src/server/rpc/procedures/weather/weather.js [hunk 6]
```diff
@@ -23,189 +21,113 @@ var isWithinMaxDistance = function(result, lat, lng) {
...
             // Return sunny if unknown
             var iconName = '01d.png';
             if (body.weather && body.weather[0]) {
                 iconName = body.weather[0].icon+'.png';
             }
-            request.get(baseIconUrl+iconName).pipe(response);
+            let queryOpts = {
+                queryString: iconName,
+                baseUrl: 'http://openweathermap.org/img/w/',
+                cache: false
```
Comment (brollb) [inline src/server/rpc/procedures/weather/weather.js]: a minor thing but these images could be cached

Comment (hamidzr) [inline src/server/rpc/procedures/weather/weather.js]: I remember why I didn't cache it, the TTL for the cache store on this service is set to 60sec and it's kinda low and since the call to get the image has no rate limit I didn't include it in the cache. but it still could be useful

**Comment by hamidzr** [comment]
the merge conflict was from #1145 which now should be merged in correctly.
